lib/compilers/tinygo.ts

```diff
@@ -0,0 +1,219 @@
+// Copyright (c) 2022, Compiler Explorer Authors
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import _ from 'underscore';
+
+import {BaseCompiler} from '../base-compiler';
+import * as utils from '../utils';
+
+import {ClangParser} from './argument-parsers';
+
+// Each arch has a list of jump instructions in
+// Go source src/cmd/asm/internal/arch.
+// x86 -> j, b
+// arm -> cb, tb
+// s390x -> cmpb, cmpub
+const jumpRe = /^(j|b|cb|tb|cmpb|cmpub).*/i;
+
+export class TinyGoCompiler extends BaseCompiler {
+    private readonly goroot: string;
+
+    static get key() {
+        return 'tinygo';
+    }
+
+    constructor(compilerInfo, env) {
+        super(compilerInfo, env);
+        this.goroot = this.compilerProps(`compiler.${this.compiler.id}.goroot`);
+        // this.goarch = this.compilerProps(`compiler.${this.compiler.id}.goarch`);
+        // this.goos = this.compilerProps(`compiler.${this.compiler.id}.goos`);
+    }
+
+    convertNewGoL(code) {
+        const re = /^\s+(0[Xx]?[\dA-Za-z]+)?\s?(\d+)\s*\(([^:]+):(\d+)\)\s*([A-Z]+)(.*)/;
+        const reUnknown = /^\s+(0[Xx]?[\dA-Za-z]+)?\s?(\d+)\s*\(<unknown line number>\)\s*([A-Z]+)(.*)/;
+        const reFunc = /TEXT\s+[".]*(\S+)\(SB\)/;
```

Comment:
Can be moved into top level + maybe give them a more descriptive name

Reply:
Will do so also for golang.js when I move it to ts, good call